### DISCUSSION TITLE
Where to put <ui:Dialog/>

### DISCUSSION THREAD

**Discussion by 2001joni** [lepoco/wpfui#640] [General]
Hello, I am new to wpf and xaml and have a question where I need to put <ui:Dialog/>

Currently it is in DashboardPage.xaml
Opening and closing works fine but the location isn't correct.

![Code](https://github.com/lepoco/wpfui/assets/75949601/e173f4bc-f451-4813-8693-47d4a1147005)
![View](https://github.com/lepoco/wpfui/assets/75949601/b8f0671d-086a-4365-a175-ec010c47753f)
![Dialog](https://github.com/lepoco/wpfui/assets/75949601/8ca0e430-a609-4333-a916-dce198bd1b5e)

**Comment by RoyMk** [comment]
The big problem here is that you are setting the property `VerticalAlignment="Top"`. This pushes all the content to the top portion of the window.  You are no longer covering the full frame area, just the top portion. Another issue is that in your `Grid.ColumnDefinitions` you are setting both columns to `Auto`, you are limiting the grid size to whatever content is in it, this will shrink to whatever space your dialog open buttons are taking. This is why your dialog opens up only above the area you have your buttons in. I would also recommend you don't add a margin to the entire base grid, instead add the margin to the control you are using. Try the code below.  

By setting the properties below, we make sure our dialog is covering all the window space and is above all other controls
```
 Grid.RowSpan="99"
 Grid.ColumnSpan="99"
 Panel.ZIndex="99"
```
```
<Grid Margin="56,0" >
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="*"/>
            <ColumnDefinition Width="*"/>
        </Grid.ColumnDefinitions>

        <ui:Button Click="BUTTONEVENT" Content="Open Dialog" />


        <ui:Dialog
            Foreground="White"
            Name="dialog1"
            Title="TestDialog"
            Grid.RowSpan="99"
            Grid.ColumnSpan="99"
            Panel.ZIndex="99"
            ButtonLeftClick="BUTTONEVENT"
            Message="Text" />
    </Grid>
</Page>
```
